### PR TITLE
truncate revs to revs limit

### PR DESCRIPTION
**Pull request by iilyak**
This PR drags unrelated commits because the sync between ASF and github is currently broken.

### CODE REVIEW

**Comment by iilyak** [comment]
The PR has been rebased

File: test/couch_db_doc_tests.erl [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    DbName = ?tempdb(),
+    config:set("couchdb", "stem_interactive_updates", "false", false),
+    {ok, Db} = couch_db:create(DbName, [?ADMIN_CTX]),
+    couch_db:close(Db),
+    DbName.
+
+
+teardown(DbName) ->
+    ok = couch_server:delete(DbName, [?ADMIN_CTX]),
+    ok.
```
Comment (jaydoane) [inline test/couch_db_doc_tests.erl]: This 'ok' could be omitted.

File: test/couch_db_doc_tests.erl [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        "CouchDB doc tests",
+        {
+            setup,
+            fun start/0, fun test_util:stop_couch/1,
+            {
+                foreach,
+                fun setup/0, fun teardown/1,
+                [
+                    fun should_truncate_number_of_revisions/1                ]
```
Comment (jaydoane) [inline test/couch_db_doc_tests.erl]: The location of the closing `]` is probably not where you want it

**Comment by jaydoane** [comment]
+1

**Comment by davisp** [comment]
+1

**Pull request merged**